### PR TITLE
blog:integration-test

### PR DESCRIPTION
**Pull request by Melaku05**
# Project requirements
## Integration specs
- User index page
- user show page
- User post index page
- Post show page

### CODE REVIEW

File: spec/integration/users_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      all('img').each do |i|
+        expect(i[:src]).to eq('https://dummyimage.com/600x400/000/fff')
+      end
+    end
+
+    it 'Shows the number of posts' do
+      all(:css, '.num_post').each do |post|
+        expect(post).to have_content('Number of posts: 0')
+      end
+    end
```
Comment (DammyShittu) [inline spec/integration/users_index_spec.rb]: - [x] For the user index page you are missing the test that redirects to the user's show page when you click on a user. Kindly include that.

File: spec/integration/users_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      click_link 'See all posts'
+      expect(page).to have_current_path user_posts_path(@melaku)
+    end
+
+    it "click see all posts and redirects to user's post's index page." do
+      click_link 'See all posts'
+      expect(page).to have_current_path user_posts_path(@melaku)
+    end
+  end
+end
```
Comment (DammyShittu) [inline spec/integration/users_show_spec.rb]: - [x] For the user show page you are missing the test that asserts you can see the user's first 3 posts. Kindly include that as well

File: spec/integration/posts_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    it 'can see the first comment on a post' do
+      visit(user_posts_path(@user.id))
+      expect(page).to have_content 'Well Done 👏👏👏'
+    end
+
+    it "redirects the user to the post's show page after clickin on it" do
+      visit(user_posts_path(@user.id))
+      click_link 'Ruby and Rails 1'
+      expect(page).to have_current_path user_post_path(@post1.author_id, @post1)
+    end
```
Comment (DammyShittu) [inline spec/integration/posts_index_spec.rb]: - [x] For the posts index you are missing the following tests:

  * Tests that asserts you can see how many comments a post has.
  * Tests that asserts you can see how many likes a post has.

File: spec/integration/posts_show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      expect(page).to have_content(comment.author.name)
+    end
+
+    it 'can see the comments of each commentor.' do
+      expect(page).to have_content 'really nice post Michael'
+      expect(page).to have_content 'Absolutely amazing 1'
+      expect(page).to have_content 'Absolutely amazing 2'
+      expect(page).to have_content 'Absolutely amazing 3'
+      expect(page).to have_content 'Absolutely amazing 4'
+    end
```
Comment (DammyShittu) [inline spec/integration/posts_show_spec.rb]: - [x] For the post show page you are missing the test that asserts you can see the post title.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi team,

## Highlights
✔️ Use of capybara for testing features
✔️ All written tests are passing

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lordkaito — APPROVED** [review]
Hi @Melaku05, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**